firehose-cloudwatch-trigger-stack.yaml

```diff
@@ -106,32 +108,50 @@ Resources:
           def lambda_handler(event, context):
               response = {}
               try:
-                  if event['RequestType'] == 'Create' or event['RequestType'] == 'Update':
-                      event_data = event['ResourceProperties']
-                      firehose_arn = event_data.get('LoggingFirehoseStreamArn', '')
-                      
-                      log_group_config_str = event_data.get('LogGroupConfig', [])
-                      invalid_log_groups = set(event_data.get('InvalidLogGroups', []))
-                      logger.info(f'Invalid Log Groups: {invalid_log_groups}')
-    
-                      # Parsing LogGroupConfig JSON array
-                      log_group_config = json.loads(log_group_config_str)
+                  event_data = event['ResourceProperties']
+                  request_type = event['RequestType']
+                  firehose_arn = event_data.get('LoggingFirehoseStreamArn', '')
+                  log_group_config_str = event_data.get('LogGroupConfig', [])
+                  invalid_log_groups = set(event_data.get('InvalidLogGroups', []))
+
+                  # Parsing LogGroupConfig JSON array
+                  log_group_config = json.loads(log_group_config_str)
+
+                  # Unique filter name for this stack using firehose ARN
+                  filter_name = f'LoggingFirehoseSubscription_{hashlib.sha256(firehose_arn.encode()).hexdigest()[:20]}'
```

Comment:
maybe add NewRelic also as a prefix to this name.

Reply:
Changed it to **NewRelicLogsFirehoseSubscription_** as per the naming convention we are following for other resources.